### DISCUSSION TITLE
bug(SelectTree): 设置初始值后无法选中对应项

### DISCUSSION THREAD

**Discussion by tiansfather** [dotnetcore/BootstrapBlazor#4823] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

与https://github.com/dotnetcore/BootstrapBlazor/issues/988相同，但最新版本9.1.3仍然复现

### Expected Behavior

`
@page "/"
@attribute [TabItemOption(Text = "Index", Closable = false)]
<SelectTree Items="BindItems" Value="@("文件一")"></SelectTree>
@code{
    private List<TreeViewItem<string>>? BindItems { get; set; } = new List<TreeViewItem<string>>();
    protected override async Task OnInitializedAsync()
    {
        await base.OnInitializedAsync();
        //没有下面这行就能显示选中值
         await Task.Yield();
         LoadData();

    }

    private void LoadData()
    {
        BindItems =
        [
          new TreeViewItem<string>("目录一")
          {
              Text ="目录一",
              Icon = "fa-solid fa-folder",
              ExpandIcon = "fa-solid fa-folder-open",
              Items =
              [
                  new TreeViewItem<string>("子目录一")
                  {
                      Text ="子目录一",
                      Icon = "fa-solid fa-folder",
                      ExpandIcon = "fa-solid fa-folder-open",
                      Items =
                      [
                          new TreeViewItem<string>("文件一") { Text = "文件一", Icon = "fa-solid fa-file" },
                          new TreeViewItem<string>("文件二") { Text = "文件二", Icon = "fa-solid fa-file" }
                      ]
                  }
              ]
          }
      ];
    }
}
`

### Interactive render mode

Interactive Server (Interactive server-side rendering (interactive SSR) using Blazor Server)

### Steps To Reproduce

![image](https://github.com/user-attachments/assets/e7a77a94-ee27-4ab4-abf6-b20a185a3aa3)
[复现工程.zip](https://github.com/user-attachments/files/18103948/demo.zip)

### .NET Version

NET8.0

**Comment by bb-auto** [comment]
@tiansfather Thank you for filing this issue. In order for us to investigate this issue, please provide [a minimalistic repro project](https://github.com/dotnetcore/bootstrapblazor/blob/main/docs/repro.md) that illustrates the problem.

**Comment by ArgoZhang** [comment]
@tiansfather 感谢感谢，我稍后复现，如果是bug 修复一下后发包

**Comment by ArgoZhang** [comment]
@tiansfather 不是 BUG 我转成讨论，我再给答案

**Comment by ArgoZhang** [comment]
@tiansfather 你的例子我看了。怎么说呢。这都是最最最基本的问题。并不是组件的 bug。可以 B 站关注一下我的视频教程。下面是链接

https://space.bilibili.com/660853738

正确代码有很多种，主要是看你设计。比如 `BindItems` 初始化给 `null`

Index.razor
```
@page "/"
@attribute [TabItemOption(Text = "Index", Closable = false)]

@if (BindItems != null)
{
    <SelectTree Items="BindItems" Value="@("文件一")"></SelectTree>
}

@code {
    private List<TreeViewItem<string>>? BindItems { get; set; }

    protected override async Task OnInitializedAsync()
    {
        await base.OnInitializedAsync();
        //没有下面这行就能显示选中值
        await Task.Yield();

        LoadData();
    }

    private void LoadData()
    {
        // 省略代码
    }
}
```

知识点是数据异步加载与组件渲染周期

**Comment by tiansfather** [comment]
那为什么Select没有这个问题呢
![image](https://github.com/user-attachments/assets/911c4634-0780-41b1-a06b-2882e8f83b34)

**Reply by ArgoZhang** [reply]
@tiansfather 实在忙的不可开交。开源项目，能否看一下源码，如果有不明白的我再给你解释，或者看完源码明白了，可以帮忙改进一下 `SelectTree`

**Comment by tiansfather** [comment]
好的，辛苦了

**Comment by tiansfather** [comment]
请问下组件里这样修改是不是就可以了，我测试是可以了
![image](https://github.com/user-attachments/assets/ce93a493-685c-4e6c-88aa-b6dc7d263898)

**Reply by ArgoZhang** [reply]
@tiansfather 帮忙跑一下单元测试，如果没问题，就来个 PR 我觉得可以

**Comment by tiansfather** [comment]
好的，提交了
